### PR TITLE
Add support for method calls using wrapNodeMethod.

### PR DESCRIPTION
**Pull request by Beanow**

### CODE REVIEW

**Comment by Beanow** [comment]
Although I may be inclined to use `wrapNode(obj.method.bind(obj))` as well.

**Comment by Avaq** [comment]
My personal preference is to omit the usage of `this` as much as possible, so I don't often run into this problem. You may need to do this to wrap some libraries, and in such cases I'd probably use `wrapNode(bind(obj.fn, obj))`, or:

```js
const wrapNodeLib = compose(wrapNode, bind(__, lib));

const libFn1 = wrapNodeLib(lib.fn1);
const libFn2 = wrapNodeLib(lib.fn2);
```

However, if you insist this function is added, I won't object, given that you provide some unit tests along with it. :)

File: src/util/future.js [hunk 1]
```diff
@@ -21,6 +21,25 @@ export const wrapNode = f => (...arg) => Future(
...
 
 /**
+ * Make a node-style async object method return a Future.
+ *
+ * Takes a method which uses a node-style callback for continuation and
+ * returns a function which returns a Future for continuation.
+ * 
+ * Note: this is slightly different from wrapNode since it accepts an object to use as 'this'.
+ *
+ * @sig wrapNodeMethod :: (x..., (a, b -> Void) -> Void) -> x... -> Future[a, b]
```
Comment (Avaq) [inline src/util/future.js]: It seems the type-signature is missing the `obj` parameter.

File: src/util/future.js [hunk 2]
```diff
@@ -21,6 +21,25 @@ export const wrapNode = f => (...arg) => Future(
...
+ *
+ * @param {object} obj The 'this' argument of the method.
+ * @param {Function} f The node method to wrap.
+ *
+ * @return {Function} A function which returns a Future.
+ */
+export const wrapNodeMethod = (obj, f) => (...arg) => Future(
```
Comment (Avaq) [inline src/util/future.js]: The two arguments here are not curried. I am of the opinion that they should be.

**Pull request closed without merging**